homeassistant/components/nsw_fuel_station/__init__.py

```diff
@@ -5,68 +5,97 @@
 from dataclasses import dataclass
 import datetime
 import logging
+from typing import TYPE_CHECKING
 
-from nsw_fuel import FuelCheckClient, FuelCheckError, Station
+import nsw_fuel
 
-from homeassistant.core import HomeAssistant
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
 from homeassistant.helpers import config_validation as cv
-from homeassistant.helpers.typing import ConfigType
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import DATA_NSW_FUEL_STATION
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from homeassistant.core import HomeAssistant
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = "nsw_fuel_station"
 SCAN_INTERVAL = datetime.timedelta(hours=1)
 
 CONFIG_SCHEMA = cv.platform_only_config_schema(DOMAIN)
 
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the NSW Fuel Station platform."""
-    client = FuelCheckClient()
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+) -> bool:
+    """Set up this integration using UI."""
+    client = nsw_fuel.FuelCheckClient()
 
     async def async_update_data():
         return await hass.async_add_executor_job(fetch_station_price_data, client)
 
-    coordinator = DataUpdateCoordinator(
-        hass,
-        _LOGGER,
-        config_entry=None,
-        name="sensor",
-        update_interval=SCAN_INTERVAL,
-        update_method=async_update_data,
-    )
-    hass.data[DATA_NSW_FUEL_STATION] = coordinator
+    if hass.data.get(DOMAIN) is None:
+        coordinator = DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            config_entry=entry,
+            name="sensor",
+            update_interval=SCAN_INTERVAL,
+            update_method=async_update_data,
+        )
+        hass.data[DOMAIN] = coordinator
+
+        await coordinator.async_config_entry_first_refresh()
 
-    await coordinator.async_refresh()
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
```

Comment:
Oh so previously we had 1 coordinator per instance? I think you could keep this. Other integrations have this pattern as well like iron_os